### DISCUSSION TITLE
Alt+` is not recognised on german keyboard

### DISCUSSION THREAD

**Discussion by itsanonmelody** [helix-editor/helix#11615] [Troubleshooting]
### Summary

Typing a simple backtick ` on a german keyboard requires three keystrokes: Shift+', Space. Helix recognises this. However, when I try to input a backtick with Alt, it doesn't work anymore. Only the backtick is recognised.

### Reproduction Steps

 

I tried this:

1. `hx`
2. `i`
3. `hElLo woRLd`
4. `<Esc>`
5. `x`
6. `Shift+Alt+', Space` / `Shift+Alt+', Alt+Space` / `Shift+Alt+', Shift+Alt+Space`

I expected this to happen:

With default keybinds, all the characters in the text `hElLo woRLd` should be uppercased: `HELLO WORLD`.

Instead, this happened:

The text is lowercased instead. Alt+\` seems to be recognised as \`.

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>

```
2023-12-12T00:14:13.891 helix_view::editor [ERROR] editor error: 1 unsaved buffer(s) remaining: ["[scratch]"]
2023-12-12T00:16:25.177 helix_view::editor [ERROR] editor error: 1 unsaved buffer(s) remaining: ["[scratch]"]
2023-12-12T00:38:46.206 helix_view::editor [ERROR] editor error: 1 unsaved buffer(s) remaining: ["[scratch]"]
```

</details>

### Platform

Arch Linux

### Terminal Emulator

kitty 0.31.0

### Installation Method

Arch extra repository through pacman

### Helix Version

Helix 23.10-1

**Comment by CptPotato** [comment]
As a workaround you could try rebinding the keys to not use <kbd>Alt</kbd>. I'm using these on a German keyboard layout:

```toml
"´" = "switch_to_lowercase"
"`" = "switch_to_uppercase"
```

**Comment by pascalkuthe** [comment]
Do these key sequences work in other tui programs (your shell, nvim)? Kitty should just forward us the keycode there isn't too me for us to do

**Comment by Brixy** [comment]
Hi,

I use this environment variable, and helix’s `Alt+`` works as expected:

```
export XKB_DEFAULT_VARIANT=nodeadkeys
```

In order to write letters with accents like `é è` etc. I use this line:

```
export XKB_DEFAULT_OPTIONS=compose:caps
```

In order to type an `è` for example you need to press:

- `Caps`
- `Shift+´`
- `e`

Also, I prefer these settings for writing code with ticks and backticks.

**Comment by itsanonmelody** [comment]
> Do these key sequences work in other tui programs (your shell, nvim)? Kitty should just forward us the keycode there isn't too me for us to do

I tried binding Alt-` to a command in Neovim and it doesn't work either. The solution from @Brixy works well as it removes the requirement to do two keystrokes to type a backtick.

**Comment by xchray** [comment]
Hi, all, thanks for discussion so far. I have same issue in Windows, is there any possible solution on Windows? 

Thanks so much.
